File: 703-kth-largest-element-in-a-stream/703-kth-largest-element-in-a-stream.java
class KthLargest {
    private static int k;
    PriorityQueue<Integer> minHeap = new PriorityQueue<>();
    
    public KthLargest(int k, int[] nums) {
        this.k = k;
        for(int num : nums) {
            minHeap.add(num);
            if(minHeap.size() > k) minHeap.poll();
        }
    }
    
    public int add(int val) {
        minHeap.add(val);
        while(minHeap.size() > k) minHeap.poll();
        return minHeap.peek();
    }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.add(val);
 */

File: 703-kth-largest-element-in-a-stream/703-kth-largest-element-in-a-stream.cpp
class KthLargest {
private:
    int K;
    priority_queue<int, vector<int>, greater<int>> minHeap;
public:
    KthLargest(int k, vector<int>& nums) {
        K = k;
        for(int num : nums) {
            minHeap.push(num);
            if(minHeap.size() > k) 
                minHeap.pop();
        }
    }
    
    int add(int val) {
        minHeap.push(val);
        while(minHeap.size() > K) 
            minHeap.pop();
        return minHeap.top();
    }
};

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */